mbtiles/src/mbtiles.rs

```diff
@@ -223,7 +277,94 @@ impl Mbtiles {
             self.check_agg_tiles_hashes(&mut conn).await
         }
     }
-
+    pub async fn statistics<T>(&self, conn: &mut T) -> MbtResult<Statistics>
+    where
+        for<'e> &'e mut T: SqliteExecutor<'e>,
+    {
+        let file_size =
+            PathBuf::from(&self.filepath).metadata().unwrap().len() as f64 / 1024.0 / 1024.0;
+        let page_size = query!("PRAGMA page_size;")
+            .fetch_one(&mut *conn)
+            .await?
+            .page_size;
+        let tile_infos_query = query!(
+            r#"SELECT
+                zoom_level AS zoom,
+                count( ) AS count,
+                min( length( tile_data ) ) / 1024.0 AS smallest,
+                max( length( tile_data ) ) / 1024.0 AS largest,
+                avg( length( tile_data ) ) / 1024.0 AS average,
+                min(tile_column) as min_tile_x,
+                min(tile_row) as min_tile_y,
+                max(tile_column) as max_tile_x,
+                max(tile_row) as max_tile_y 
+            FROM tiles
+            GROUP BY zoom_level"#
+        );
+        let mbt_type = self.detect_type(&mut *conn).await?;
+        let level_rows = tile_infos_query.fetch_all(&mut *conn).await?;
+        let mut level_details: Vec<LevelDetail> = level_rows
+            .into_iter()
+            .map(|r| {
+                let zoom = r.zoom.unwrap() as u8;
+                let count = r.count as u32;
+                let tile_length = 40075016.7 / (2_u32.pow(zoom as u32)) as f64;
+
+                let smallest = r.smallest.unwrap_or(0.0) as f32;
+                let largest = r.largest.unwrap_or(0.0) as f32;
+                let average = r.average.unwrap_or(0.0) as f32;
+
+                let min_tile_x = r.min_tile_x.unwrap();
+                let min_tile_y = r.min_tile_y.unwrap();
+                let max_tile_x = r.max_tile_x.unwrap();
+                let max_tile_y = r.max_tile_y.unwrap();
+
+                let minx = -20037508.34 + min_tile_x as f64 * tile_length;
+                let miny = -20037508.34 + min_tile_y as f64 * tile_length;
+                let maxx = -20037508.34 + (max_tile_x as f64 + 1.0) * tile_length;
+                let maxy = -20037508.34 + (max_tile_y as f64 + 1.0) * tile_length;
+
+                let bbox = Bounds::new(minx, miny, maxx, maxy);
+                LevelDetail {
+                    zoom: format!("{zoom}"),
+                    count,
+                    smallest,
+                    largest,
+                    average,
+                    bounding_box: bbox,
+                }
+            })
+            .collect();
+        let details_of_all = LevelDetail {
```

Comment:
summary of all levels will probably have to be a different struct, or better yet - just put those values in the `Statistics` struct itself. The values will be different because
* most of them will be `Option<...>` -- if there is no data, they will be set to None
* bbox might not be needed because most of the time tileset will contain zoom-0 tile, which covers the whole eart. bbox is mostly useful per zoom.
* there is no `zoom` value - so you should keep the `zoom` value inside LevelDetail as `u8`, not `String`.